source/iopipe/json/serialize.d

```diff
@@ -845,118 +836,511 @@ unittest
     assert(cArr.length == 3);
 }
 
-void serializeImpl(T, Char)(scope void delegate(const(Char)[]) w, ref T val) if (__traits(isStaticArray, T))
+//alias OutputDelegate(Char) = void delegate(const(Char)[]);
+
+struct FormattingOptions
+{
+    bool lineBreaks;
```

Comment:
I want to be able to specify line breaks for objects and arrays separately. For instance, in dub.json, I want arrays of strings to come out inline, but objects to be indented. This is something I did in my branch.

Reply:
How should it behave when serializing an object inside the array?

Reply:
Between object members, there would be newlines, between array members, there would not.
so...

```json
{
    "foo" : [{
        "bar" : 5,
        "baz" : "hello"
    }, {
        "bar" : 8,
        "baz" : "world"
    }]
}

Reply:
Admittedly, if there are mixed types in the array it's going to look odd. But for the most part this should be fine.